File: examples/reduced_annotations/c_language/partially_annotated1.c
#include "partially_annotated1.h"
#include "partially_annotated2.h"

int not_annotated_inc_h_1(int i)
{
/*@ assert false; @*/ 
  return not_annotated_inc_h_2(i); 
}
void not_annotated_c_1_1(){/*@ assert false; @*/}

int annotated_inc_1_1(int i)
  //@ requires i < 100;
  //@ ensures result == i + 1;
{
  int j = 1;
  j = annotated_inc_1_2(j);
  //@ assert j == 2;
  j = annotated_inc_2_1(j);
  //@ assert j == 3;
  j = annotated_inc_2_2(j);
  //@ assert j == 4;
  return ++i;
}

void not_annotated_c_1_2(){/*@ assert false; @*/}

int annotated_inc_1_2(int i)
  //@ requires i < 100;
  //@ ensures result == i + 1;
{
  return ++i;
}

void not_annotated_c_1_3(){/*@ assert false; @*/}

File: examples/reduced_annotations/java_language/PartiallyAnnotated1.java
class NotAnnotatedClass1_1
{
  private int x; 

  NotAnnotatedClass1_1(int i) {/*@ assert false; @*/x = i;}

  int NotAnnotatedClass1_1_1(int i) {/*@ assert false; @*/return i;}
}

public class PartiallyAnnotated1 implements PartiallyAnnotatedInterface1
{
  //@ predicate PartiallyAnnotatedClassPred1(int value) = this.i |-> value;
  
  private int i;
  
  public PartiallyAnnotated1()
  {
    this.i = 0;
  }
  
  public PartiallyAnnotated1(int i)
    //@ requires true;
    //@ ensures  PartiallyAnnotatedClassPred1(i);
  {
    this.i = i;
    //@ close PartiallyAnnotatedClassPred1(i);
  }
  
  public int annotatedIncInterface1_1(int i)
    //@ requires i < 100;
    //@ ensures result == i + 1;
  {
    return ++i;
  }

//   class NotAnnotatedClass1_2
//   {
//     private int x; 
// 
//     NotAnnotatedClass1_3(int i) {/*@ assert false; @*/x = i;}
// 
//     int NotAnnotatedClass1_3_1(int i) {/*@ assert false; @*/return i;}
//   }

  public int NotAnnotatedIncInterface1(int i) 
  {
    //@ assert false;
    PartiallyAnnotatedInterface2 o = new PartiallyAnnotated2();
    return o.NotAnnotatedIncInterface2(i);
  }

  public int annotatedIncInterface1_2(int i)
    //@ requires i < 100;
    //@ ensures result == i + 1;
  {
    return ++i;
  }
  
  public int annotatedIncClass1(int i)
    //@ requires i < 100;
    //@ ensures result == i + 1;
  {
    return ++i;
  }
}

class NotAnnotatedClass1_3
{
  private int x; 

  NotAnnotatedClass1_3(int i) {/*@ assert false; @*/x = i;}

  int NotAnnotatedClass1_3_1(int i) {/*@ assert false; @*/return i;}
}
